src/fundamental/macro-fundamental.h

```diff
@@ -305,3 +305,10 @@ static inline size_t ALIGN_TO(size_t l, size_t ali) {
                 (l <= SIZE_MAX - (ali - 1)),      /* overflow? */      \
                 ((l) + (ali) - 1) & ~((ali) - 1),                      \
                 VOID_0)
+
+#define UPDATE_FLAG(orig, flag, b)                      \
+        ((b) ? ((orig) | (flag)) : ((orig) & ~(flag)))
```

Comment:
one of those days we should figure out how to express this double eval free

Reply:
Turns out, it actually already is (thanks to ternary op).

Reply:
oh, indeed